### PR TITLE
Update kfto test to utilise alpaca dataset image to access dataset instead of downloading it from huggingface

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
## Description

[RHOAIENG-16198](https://issues.redhat.com/browse/RHOAIENG-16198)

## How Has This Been Tested?

- Update test to utilise alpaca dataset image to access dataset instead of downloading it from huggingface
- Updated training script to use sequential dataset pre-processing instead of parallel to avoid MD5 hash checks

Same approach tested locally using data copied from [alpaca-dataset image](quay.io/ksuta/alpaca-dataset@sha256:2e90f631180c7b2c916f9569b914b336b612e8ae86efad82546adc5c9fcbbb8d) : 

image

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/kfto/core/kfto_training_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 )
 
 func TestPyTorchJobWithCuda(t *testing.T) {
-	runKFTOPyTorchJob(t, GetCudaTrainingImage(), "nvidia.com/gpu", 1)
+	runKFTOPyTorchJob(t, GetCudaTrainingImage(), "nvidia.com/gpu", 0)
```
Comment (ChughShilpa) [inline tests/kfto/core/kfto_training_test.go:34]: Why you have set it gpu to 0?

Comment (abhijeet-dhumal) [reply tests/kfto/core/kfto_training_test.go:34]: Ahh, my mistake..
Thank you Shilpa for pointing that out !!

File: tests/kfto/core/hf_llm_training.py [hunk 2]
```diff
@@ -77,22 +77,20 @@ def load_and_preprocess_data(dataset_dir, transformer_type, tokenizer):
 
     file_path = os.path.realpath(dataset_dir)
 
-    if transformer_type != AutoModelForImageClassification:
-        dataset = load_from_disk(file_path)
+    dataset=load_dataset('json',data_files=f'{file_path}/alpaca_data_hundredth.json')
```
Comment (sutaakar) [inline tests/kfto/core/hf_llm_training.py:80]: Can you provide the dataset file as parameter in container command?

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by Fiona-Waters — APPROVED** [review]
/lgtm
/approve
Thanks @abhijeet-dhumal

**Review by ChughShilpa — APPROVED** [review]

**Pull request merged**